### DISCUSSION TITLE
The view template "view.html.php" for areabrick does not exist

### DISCUSSION THREAD

**Discussion by iidaloiki** [pimcore/pimcore#7843] [Q&A / Support] [answered]
Hi!

I'm trying to go through Pimcore Academy to get to know Pimcore and earn certificates. Unfortunately course videos are a little outdated, so I've had some issues with things not working like they are guided.

I'm trying to make an Areablock and use it in a Document. I've created it succesfully, but when adding it to Document, I get an error saying that view template doesn't exist. Problem is that it tries to search for PHP template instead of Twig template, which I have.

Is there a way to tell Pimcore to search for Twig templates instead of PHP templates, since they are going to be deprecated in the future?

My Symfony version is 4.4.17 and Pimcore version is 6.8.5, if that helps.

**Comment by NiklasBr** [comment, marked as answer]
In your AppBundle's Controller method onKernelController:

```php
    /**
     * {@inheritDoc}
     */
    public function onKernelController(FilterControllerEvent $event)
    {
        // set auto-rendering to twig
        $this->setViewAutoRender($event->getRequest(), true, 'twig');
    }
```

**Reply by fashxp** [reply]
hi, 
for bricks see https://pimcore.com/docs/pimcore/current/Development_Documentation/Documents/Editables/Areablock/Bricks.html#page_Brick-template-auto-discovery

BR
Christian

**Reply by iidaloiki** [reply]
Thank you for quick responses!

It seems that setAutoViewRender is deprecated, so I tried @fashxp solution. Following code is what I came up with:

```
<?php

namespace AppBundle\Document\Areabrick;

use Pimcore\Extension\Document\Areabrick\AbstractTemplateAreabrick;

class Academy extends AbstractTemplateAreabrick {

  const TEMPLATE_LOCATION_GLOBAL = 'global';
  const TEMPLATE_LOCATION_BUNDLE = 'bundle';

  const TEMPLATE_SUFFIX_PHP  = 'html.php';
  const TEMPLATE_SUFFIX_TWIG = 'html.twig';

  /**
   * Determines if template should be auto-located in area bundle or in app/Resources
   *
   * @return string
   */
  public function getTemplateLocation() {
    return $this->TEMPLATE_LOCATION_GLOBAL;
  }

  /**
   * Returns view suffix used to auto-build view names
   *
   * @return string
   */
  public function getTemplateSuffix() {
    return $this->TEMPLATE_SUFFIX_TWIG;
  }
  
  public function getName() {
    return "Test Academy";
  }

  public function getDescription() {
    return "This is a description.";
  }
}
```

... But now I get error "No engine is able to work with the template "Areas/academy/view."." Did I do something wrong?

**Reply by NiklasBr** [reply]
You access the constant like this:
```php
  public function getTemplateSuffix() {
    return $this::TEMPLATE_SUFFIX_TWIG;
  }
 ```

But this is also going to be deprecated soon I suspect due to the fact that PHP templating is being removed and therefore making Twig the default template in Pimcore 7/X. No choice but to use deprecated functions while this is happening.

**Reply by iidaloiki** [reply]
Thank you for help, now it works perfectly! Good point with deprecation, hopefully things will get simpler with version 7.
